File: Task2/task2.rb
numbers = gets.to_i
max_flower = 0
if numbers <= 0
  print "Быть не может"
  exit
else
  clumba = []
  numbers.times do
    flower = gets.to_i
    if flower > 0
      clumba << flower
    end
    max_flower = flower if flower > max_flower
  end
  poliv = gets.to_i
  if poliv < max_flower
    print "Как купишь новую лейку - приходи."
    exit
  else
    step = 0
    perem = poliv
    clumba.each_with_index do |flower, i|
      if perem < flower
        step += (i * 2 + 1)
        perem = poliv
      else
        step += 1
      end
      perem -= flower
    end
    puts step
  end
end

File: Task2/task2.cpp
#include <iostream>
#include <vector>
using namespace std;
int main()
{
    setlocale(LC_ALL, "RUS");
    int numbers, max = 0;
    cin >> numbers;
    if (numbers <= 0){
            cout << "Быть не может";
            return 0;
    }
    vector<int> clumba;
    for (int i = 0; i < numbers; ++i) {
        int flower;
        cin >> flower;
        if (flower > 0)
        {
            clumba.push_back(flower);
        }
        if (flower > max)
        {
            max = flower;
        }
    }
    int poliv;
    cin >> poliv;
    if (poliv < max)
    {
        cout << " Как купишь новую лейку - приходи.";
        return 0;
    }
    int step = 0;
    int perem = poliv;
    for (int i = 0; i < clumba.size();i++)
    {
        if (perem < clumba[i])
        {
            step = step + (i * 2 + 1);
            perem = poliv;
        }
        else
        {
            step += 1;
        }
        perem -= clumba[i];
    }
    cout << step;
}
